mysql-test/suite/innodb/t/alter_rename_files.test

```diff
@@ -11,7 +11,7 @@ SET GLOBAL innodb_log_checkpoint_now=TRUE;
 # Start an ALTER TABLE and stop it before renaming the files
 SET DEBUG_SYNC='commit_cache_rebuild SIGNAL ready WAIT_FOR finish';
 
---send ALTER TABLE t1 ADD PRIMARY KEY(x)
+--send ALTER TABLE t1 FORCE;
```

Comment:
No, the table won't be rebuilt after this patch because it actually has a primary key. https://github.com/MariaDB/server/blob/9c7299365f04faf1b3963a2d891c8cb6dda89e4a/sql/table.cc#L2473

Instant index rename will happen here.

As I understand, the idea of the test is to rebuild an index. And I fixed it by making the simplest `ALTER TABLE` which always rebuilds an index.

Reply:
According to `SHOW CREATE TABLE` the table did not have a `PRIMARY KEY`. It had a `UNIQUE KEY` on the column, which could be internally treated as a primary key by InnoDB.

If the purpose of this change is only to optimize away DROP/ADD INDEX pairs into RENAME INDEX, this ‘merging’ of `UNIQUE KEY` and `PRIMARY KEY` seems to be an undocumented regression.

Reply:
> According to SHOW CREATE TABLE the table did not have a PRIMARY KEY.

Yes, but `TABLE::TABLE_SHARE::primary_key != MAX_KEY`. It looks like an implicit `PRIMARY KEY`. And InnoDB threats this like a real `PRIMARY KEY` inside `ha_innobase::create()`. And this patch optimizes this `ALTER` to index rename.

Here is this feature is described https://jira.mariadb.org/browse/MDEV-6794